### ISSUE TITLE
[Feature] Add an option to disable "crash detection"

### ISSUE THREAD

**Issue by LoganDark** [BetterDiscord/BetterDiscord#1304]
**Is your feature request related to a problem? Please describe.**
My Discord client often crashes due to unrelated Windows issues (i.e. ETW traces, video driver resets, some types of hangs), and each time it does, BetterDiscord confidently asserts "It seems like BetterDiscord has crashed your Discord client. Bye bye!"

BetterDiscord is never the culprit of course, but it always assumes that it was the source of my crash, and disables itself, requiring me to restart Discord *again* in order to re-enable it.

**Describe the feature you'd like**
One possibility would be improving the accuracy of this detection so that non-BetterDiscord-related crashes *won't* disable BetterDiscord, but that's incredibly risky with the possibility of false-negatives and whatnot. So instead I'd like to propose a second option: **expose a toggle in the settings that allows me to turn off this detection**.

If BetterDiscord is crashing my client, **I as a developer know the location of my plugins and know how to fix the damage**. I would like the ability to acknowledge that fact and tell BetterDiscord "yes, I do know what I'm doing", just like how it is already possible to (for example) re-enable DevTools.

This would improve my experience since this detection has never had a true positive for me and my course of action after each one is always to restart Discord again because BetterDiscord does not know what it's talking about. I would love to just cut to the chase and have BetterDiscord stay enabled after any crash.

**Additional context**
N/A

**Comment by LoganDark** [comment]
This wasn't supposed to submit, hold on. I'm still filling out the form, I'll edit this when done.

**Comment by QbDesu** [comment]
Simply reloading with Ctrl+R is enough to reenable BetterDiscord. The feature is important to prevent BetterDiscord from crash looping your client and not rendering your client unusable if something broke.

(The most common cause of sporadic client crashes are OOM crashes from poorly written plugins btw.)

**Comment by LoganDark** [comment]
> The feature is important to prevent BetterDiscord from crash looping your client and not rendering your client unusable if something broke.

I know that in the general case where it's actually caused by BD, this feature is important, but that has never been the case for me. That's why this would be an "I know what I'm doing and I know it's risky" sort of thing.

> (The most common cause of sporadic client crashes are OOM crashes from poorly written plugins btw.)

That can't be right, I have 40GB of RAM, lol!

**Comment by SmolAlli** [comment]
memory leaks can still lead to running out of memory. so, yes i can be the case. not lol.

**Comment by LoganDark** [comment]
> memory leaks can still lead to running out of memory. so, yes i can be the case. not lol.

I'm pretty sure I would notice if Discord started taking up all my memory. I open Task Manager at least once per day and Discord is open wayy longer than that.

That's completely besides the point of this issue.

**Comment by ggtylerr** [comment]
In my experience Discord has a limit of around 1.5 GB. At that point it "runs out" and crashes.

**Comment by LoganDark** [comment]
I would like to stay on-topic please.

**Comment by Maverynthia** [comment]
I'd like the option to at least have it reload discord with my plugins enabled. I do like the idea of turning it off. Or maybe have a timer that if it crashes x times in y time then it disables plugins, else just reload.

**Comment by LoganDark** [comment]
It looks like the rationale for dismissing this issue is that there could be some hidden memory leak or bug in my Discord client, making this not a false positive; but even if there is, the stability of my client *the majority of the time* means that there's no reason to panic and disable BetterDiscord every time it crashes. So a toggle would still be very helpful and appreciated.

**Comment by LoganDark** [comment]
@rauenzi I don't consider this issue resolved by a "Try Again" button that can be clicked only after BetterDiscord has already been disabled. Could you re-open this until a proper setting is added to disable crash detection *before* the fact?

**Comment by LoganDark** [comment]
Should I open a new issue for the permanent setting? I'm not familiar with how you prefer to operate BD's issue tracker.
